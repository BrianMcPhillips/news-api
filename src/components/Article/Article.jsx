import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ 
  urlToImage,
  title,
  description,
  author,
  url
}) => (
  <div>
    <img src={urlToImage} alt={title} />
    <h2>{title}</h2>
    <h4>{description}</h4>
    <p>{author}</p>
    <a href={url}>View Article</a>
  </div>
);

Article.propTypes = {
  urlToImage: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Article;
