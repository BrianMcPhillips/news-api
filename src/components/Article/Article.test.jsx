import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  afterEach(() => cleanup());
  
  it('renders a single article', () => {
    const { asFragment } = render(<Article 
      urltoImage="http://placeholder.com/villager"
      title="Best article ever"
      description="This is beyond a great article so insightful"
      author="test testerson"
      url="www.test.test.com"  
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});

