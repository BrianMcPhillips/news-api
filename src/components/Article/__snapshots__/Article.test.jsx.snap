// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders a single article 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="Best article ever"
      src="http://placeholder.com/villager"
    />
    <h2>
      Best article ever
    </h2>
    <h4>
      This is beyond a great article so insightful
    </h4>
    <p>
      test testerson
    </p>
    <a
      href="www.test.com"
    >
      View Article
    </a>
  </div>
</DocumentFragment>
`;
